src/contracts/libraries/EscrowBits.sol

```diff
@@ -58,9 +58,9 @@ library EscrowBits {
         return (result & _FULL_REFUND) != 0;
     }
 
-    function updateEscrow(uint256 result) internal pure returns (bool) {
-        // Only update solver escrow if they need to refund gas
-        // returns true if solver has to do the refund.
-        return (result & _NO_REFUND == 0);
+    function bundlersFault(uint256 result) internal pure returns (bool) {
+        // Only update solver escrow if failure is not due to bundler's fault
+        // returns true if bundler blamed and no solver refund required
+        return (result & _NO_REFUND != 0);
```

Comment:
This looks good to me, I agree this is more readable and equivalent

FYI one insight (that was present in the old code too) is that it's possible for one `result` to have errors across multiple categories, and this code has been giving more "precedence" to the "bundler's fault" errors, which I think is correct. For example if `result` has errors in `_NO_REFUND` and also `_PARTIAL_REFUND`:

Old code: `result & _NO_REFUND == 0` is FALSE. This means `updateEscrow()` returns FALSE. This means `GasAccounting.sol` enters `_setWriteoffs()`. Ultimately the "bundler's fault" error took precedence

New code: `result & _NO_REFUND != 0` is TRUE. This means `bundlersFault()` returns TRUE. This means `GasAccounting.sol` enters `_setWriteoffs()`. Ultimately the "bundler's fault" error took precedence

Reply:
Yeah this is a good observation. I think its fine as its equivalent to what it was before, but worth double checking this is intended with @thogard785

Reply:
Got it I think that makes sense, plus I see now this ties into one of the last issues here: https://github.com/spearbit-audits/review-fastlane/issues/153#issuecomment-2225335543 because it's important that the `GasPriceOverCap` error (`_PARTIAL_REFUND`) has lower priority than the `GasPriceBelowUsers` error (`_NO_REFUND`)